src/ui/deleteAccount/ViewDeleteAccount.qml

```diff
@@ -11,13 +11,40 @@ import components.forms 0.1
 import components.inAppAuth 0.1
 
 Item {
+    property bool _menuVisible: false
+    property string prevState: ""
+
     id: viewDeleteAccount
 
     Loader {
         id: loader
+        property bool isDeleteAccountAuth: true
 
-        asynchronous: true
         anchors.fill: parent
+
+        asynchronous: true
+        onStatusChanged: {
+            if (loader.status !== Loader.Ready) {
+                return;
+            }
+            
+            if (
+                prevState === ""
+                && viewDeleteAccount.state === "StateInitializing"
+            ) {
+                // `StateInitializing` is set for the first time
+                VPN.requestDeleteAccount();
+            } else if (
+                prevState !== ""
+                && viewDeleteAccount.state === "StateInitializing"
+            ) {
+                // `StateInitializing` is set during the account deletion flow,
+                // which means the user aborted the operation
+                settingsStackView.pop();
```

Comment:
can we add this where we call `VPN.cancelAccountDeletion()`?

Reply:
Created a dedicated function `cancelAccountDeletion` that is being called from the account deletion flow templates when the user cancels the operation.